### PR TITLE
Fix #1056: Remove extra method generated by "x-ms-pageable" custom nextlink operation

### LINKED ISSUE #1056: Remove extra method generated by "x-ms-pageable" custom nextlink operation
For example we have [this](https://github.com/Azure/autorest.testserver/blob/master/swagger/paging.json#L532-L573) customize next link operation `NextFragment` defined in the swagger. It generates [NextFragment()](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestServerProjects/paging/Generated/PagingRestClient.cs#L1113) operation which is correct but it also generates [NextFragmentNextPage()](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestServerProjects/paging/Generated/PagingRestClient.cs#L2200) operation which I think is not needed. We can consider removing it if there is no usecase of this extra operation.

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Output/Builders/ClientBuilder.cs [hunk 1]
```diff
@@ -85,6 +85,11 @@ public static IEnumerable<PagingMethod> BuildPagingMethods(OperationGroup operat
                     RestClientMethod method = restClient.GetOperationMethod(serviceRequest);
                     RestClientMethod? nextPageMethod = restClient.GetNextOperationMethod(serviceRequest);
 
+                    if (method == nextPageMethod)
```
Comment (pakrym) [inline src/AutoRest.CSharp/Common/Output/Builders/ClientBuilder.cs:88]: This check doesn't seem to be correct at first look. Why do we need to skip methods where get next is the same as get first?

File: src/AutoRest.CSharp/DataPlane/Output/DataPlaneRestClient.cs [hunk 2]
```diff
@@ -56,46 +43,44 @@ public DataPlaneRestClient(OperationGroup operationGroup, BuildContext<DataPlane
 
         protected override Dictionary<ServiceRequest, RestClientMethod> EnsureGetNextPageMethods()
```
Comment (pakrym) [inline src/AutoRest.CSharp/DataPlane/Output/DataPlaneRestClient.cs:44]: How different are the override and the base implemenataion?

Comment (chamons) [reply src/AutoRest.CSharp/DataPlane/Output/DataPlaneRestClient.cs:44]: Diffing suggests no difference. Collapsing.

**Comment by chamons** [comment]
So after digging some more, with a test case, Pavel and I think this is a deeper issue. We should be having paging.NextLinkOperation defined as non-null in my test case but it isn't, possibly because the code model has a recursive definition of itself.

**Comment by chamons** [comment]
Closing in favor of https://github.com/Azure/autorest.csharp/pull/1303

**Pull request closed without merging**